packages/vite/src/module-runner/moduleCache.ts

```diff
@@ -1,139 +1,110 @@
-import { isWindows, slash, withTrailingSlash } from '../shared/utils'
+import {
+  cleanUrl,
+  isWindows,
+  slash,
+  unwrapId,
+  withTrailingSlash,
+} from '../shared/utils'
 import { SOURCEMAPPING_URL } from '../shared/constants'
-import { decodeBase64 } from './utils'
+import { decodeBase64, posixResolve } from './utils'
 import { DecodedMap } from './sourcemap/decoder'
-import type { ModuleCache } from './types'
+import type { ResolvedResult } from './types'
 
 const MODULE_RUNNER_SOURCEMAPPING_REGEXP = new RegExp(
   `//# ${SOURCEMAPPING_URL}=data:application/json;base64,(.+)`,
 )
 
-export class ModuleCacheMap extends Map<string, ModuleCache> {
-  private root: string
-
-  constructor(root: string, entries?: [string, ModuleCache][]) {
-    super(entries)
-    this.root = withTrailingSlash(root)
-  }
-
-  normalize(fsPath: string): string {
-    return normalizeModuleId(fsPath, this.root)
+export class ModuleRunnerNode {
+  public importers = new Set<string>()
+  public imports = new Set<string>()
+  public lastInvalidationTimestamp = 0
+  public evaluated = false
+  public meta: ResolvedResult | undefined
+  public promise: Promise<any> | undefined
+  public exports: any | undefined
+  public file: string
+  public map: DecodedMap | undefined
+
+  constructor(
+    public id: string,
+    public url: string,
+  ) {
+    this.file = cleanUrl(id)
   }
+}
 
-  /**
-   * Assign partial data to the map
-   */
-  update(fsPath: string, mod: ModuleCache): this {
-    fsPath = this.normalize(fsPath)
-    if (!super.has(fsPath)) this.setByModuleId(fsPath, mod)
-    else Object.assign(super.get(fsPath)!, mod)
-    return this
-  }
+export class ModuleRunnerGraph {
+  private root: string
 
-  setByModuleId(modulePath: string, mod: ModuleCache): this {
-    return super.set(modulePath, mod)
-  }
+  public idToModuleMap = new Map<string, ModuleRunnerNode>()
+  public fileToModuleMap = new Map<string, ModuleRunnerNode[]>()
```

Comment:
This should be named `fileToModulesMap`. Should we also use `Set<ModuleRunnerNode>` rather than `ModuleRunnerNode[]` for consistency with `EnvironmentModuleGraph`?